Gems/Atom/Tools/AtomToolsFramework/Code/Source/Inspector/PropertyWidgets/PropertyStringBrowseEditCtrl.cpp

```diff
@@ -105,29 +105,48 @@ namespace AtomToolsFramework
         }
     }
 
-    void PropertyStringBrowseEditCtrl::Edit()
+    void PropertyStringBrowseEditCtrl::EditValue()
     {
-        Q_EMIT ValueChanged();
     }
 
-    void PropertyStringBrowseEditCtrl::Clear()
+    void PropertyStringBrowseEditCtrl::ClearValue()
     {
         m_browseEdit->setText("");
         m_browseEdit->lineEdit()->clearFocus();
-        Q_EMIT ValueChanged();
     }
 
     void PropertyStringBrowseEditCtrl::SetValue(const AZStd::string& value)
     {
         m_browseEdit->setText(value.c_str());
-        Q_EMIT ValueChanged();
     }
 
     AZStd::string PropertyStringBrowseEditCtrl::GetValue() const
     {
         return m_browseEdit->text().toUtf8().constData();
     }
 
+    void PropertyStringBrowseEditCtrl::OnValueChanged()
+    {
+        AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast(
+            &AzToolsFramework::PropertyEditorGUIMessages::RequestWrite, this);
+        AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast(
+            &AzToolsFramework::PropertyEditorGUIMessages::Bus::Handler::OnEditingFinished, this);
+    }
+
+    void PropertyStringBrowseEditCtrl::OnTextEditingFinished()
+    {
+        // This check is compensating for what might be a bug in the browser widget. If the line edit widget is read only and double clicked
+        // then the browse edit widget will send the signal that the edit button has been pressed. It's likely treating the entire read only
+        // widget as a button for convenience, UX, feedback. However, double clicking the read only line edit widget is also sending a
+        // conflicting signal that editing is finished, even though it never began. So, double clicking the widget to open the dialog
+        // triggers an erroneous value change and causes the property editor to refresh just before editing begins, placing it in a bad
+        // state.
```

Comment:
Thanks for the explanation here. That is super odd, I'd like to dive into that later when time allows, this should do the trick though.